### PR TITLE
DR2-1210 Create bagit json.

### PR DESCRIPTION
**Pull request by MancunianSam**
This replaces the three metadata files with one single json file.

I also added circe because upickle needed far too many implicits and it was complaining about the case objects. I think there is a solution but circe just works out of the box.

Also, we're adding parent ID as an optional UUID now, the parent path will be constructed in the ingest mapper.

### CODE REVIEW

File: src/test/scala/uk/gov/nationalarchives/LambdaTest.scala [hunk 1]
```diff
@@ -58,6 +58,7 @@ class LambdaTest extends AnyFlatSpec with BeforeAndAfterEach {
   val event: SQSEvent = createEvent(packageAvailable.asJson.printWith(Printer.noSpaces))
 
   private def read[T](jsonString: String)(implicit enc: Decoder[T]): T = {
+    val d = decode[T](jsonString)
```
Comment (techncl) [inline src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:61]: Is this supposed to be here?

**Review by techncl — COMMENTED** [review]
Looks good; just some minor comments.

File: src/main/scala/uk/gov/nationalarchives/FileProcessor.scala [hunk 2]
```diff
@@ -64,55 +65,31 @@ class FileProcessor(
...
+    val folderId = uuidGenerator()
     val assetId = uuidGenerator()
-    val assetPath = s"$folderPath/$assetId"
-    val folderMetadataRow = Row(NonEmptyList(folderPath.toString, List("", cite, title)))
-    val assetMetadataRow = Row(NonEmptyList(assetId.toString, List(folderPath.toString, title)))
-    val fileRow = Row(
-      NonEmptyList(fileInfo.id.toString, List(assetPath, fileInfo.fileName, fileInfo.fileSize.toString, title))
+    val folderMetadataObject = BagitMetadataObject(folderId, None, title, ArchiveFolder, Option(cite), None)
+    val assetMetadataObject = BagitMetadataObject(assetId, Option(folderId), title, Asset)
+    val fileObject =
```
Comment (techncl) [inline src/main/scala/uk/gov/nationalarchives/FileProcessor.scala:72]: Maybe `fileRowMetadataObject`?

File: src/test/scala/uk/gov/nationalarchives/LambdaTest.scala [hunk 3]
```diff
@@ -65,14 +55,18 @@ class LambdaTest extends AnyFlatSpec with BeforeAndAfterEach {
...
+    decode[T](jsonString).toOption.get
+  }
 
   private def runLambdaAndReturnStepFunctionRequest(metadataJsonOpt: Option[String] = None) = {
     stubAWSRequests(inputBucket, metadataJsonOpt = metadataJsonOpt)
     IngestParserTest().handleRequest(event, null)
 
     val sfnEvent = sfnServer.getAllServeEvents.asScala.head
-    read[SFNRequest](sfnEvent.getRequest.getBodyAsString)
+    decode[SFNRequest](sfnEvent.getRequest.getBodyAsString).toOption.get
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:70]: Good spot yeah, I've changed it.

File: src/test/scala/uk/gov/nationalarchives/FileProcessorTest.scala [hunk 4]
```diff
@@ -199,46 +199,41 @@ class FileProcessorTest extends AnyFlatSpec with MockitoSugar with TableDrivenPr
       val folderId = uuids.head
       val assetId = uuids.last
       val folderString =
-        s"""identifier,parentPath,name,title
-           |$folderId,,TEST-CITE,
-           |""".stripMargin
+        s"""{"identifier":"$folderId","parentPath":"","title":"","type":{"ArchiveFolder":{}},
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/FileProcessorTest.scala]: I think it's better to encode the `BagitMetadataObject` I've change it to create those and then print the list as a json string.

File: src/test/scala/uk/gov/nationalarchives/LambdaTest.scala (hunk 1 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:61]: Oops :smile:

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**